specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2023-04-01-preview/TenantActivityLogAlerts.json

```diff
@@ -0,0 +1,547 @@
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The request succeeded.",
+            "schema": {
+              "$ref": "#/definitions/TenantAlertRuleList"
+            }
+          },
+          "default": {
+            "description": "An error occurred and the list of Activity Log Alert rules could not be retrieved.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "TenantAlertRuleList": {
+      "description": "A list of Tenant Activity Log Alert rules.",
+      "properties": {
+        "value": {
+          "description": "The list of Tenant Activity Log Alert rules.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/TenantActivityLogAlertResource"
+          },
+          "x-ms-identifiers": [
+            "id"
+          ]
+        },
+        "nextLink": {
+          "description": "Provides the link to retrieve the next set of elements.",
+          "type": "string"
+        }
+      }
+    },
+    "TenantActivityLogAlertResource": {
+      "type": "object",
+      "description": "A Tenant Activity Log Alert rule resource.",
+      "required": [
+        "properties"
+      ],
+      "allOf": [
+        {
+          "$ref": "#/definitions/AzureResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "The Activity Log Alert rule properties of the resource.",
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/AlertRuleProperties"
+        }
+      }
+    },
+    "AlertRuleProperties": {
+      "description": "An Azure Activity Log Alert rule.",
+      "type": "object",
+      "properties": {
+        "tenantScope": {
+          "description": "The tenant GUID. Must be provided for tenant-level and management group events rules.",
+          "type": "string"
+        },
+        "scopes": {
+          "description": "A list of resource IDs that will be used as prefixes. The alert will only apply to Activity Log events with resource IDs that fall under one of these prefixes. This list must include at least one item.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "condition": {
+          "description": "The condition that will cause this alert to activate.",
+          "$ref": "#/definitions/AlertRuleAllOfCondition"
+        },
+        "actions": {
+          "description": "The actions that will activate when the condition is met.",
+          "$ref": "#/definitions/ActionList"
+        },
+        "enabled": {
+          "description": "Indicates whether this Activity Log Alert rule is enabled. If an Activity Log Alert rule is not enabled, then none of its actions will be activated.",
+          "type": "boolean",
```

Comment:
>boolean [](http://example.com/codeflow?start=19&length=7)

ARM recommends string enums over boolean. Generally enums make better properties than booleans. They are more descriptive and flexible while being easier for customers to discover, understand and use. Ref: [https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html?q=boolean#common-issues-found-in-review](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html?q=boolean#common-issues-found-in-review). They allow values other than just true/false and support description strings. Even if you only ever want two values, you can have a string enum with True/False (or Enabled/Disabled, etc.).

Reply:
This resource type is an extension resource flavor of microsoft.insights/activityLogAlerts - to make customers use alerts without owning a subscription.
Therefore, we would like the payload to be similar to it.
More specifically, identical to this API version https://github.com/Azure/azure-rest-api-specs/pull/22575/files